core/src/common.cpp

```diff
@@ -237,6 +361,44 @@ namespace irods::http
 		// Verify the bearer token is known to the server. If not, return an error.
 		auto mapped_value{irods::http::process_stash::find(bearer_token)};
 		if (!mapped_value.has_value()) {
+			// If we're running as a protected resource, assume we have a OIDC token
+			if (irods::http::globals::oidc_configuration().at("run_as_protected_resource").get<bool>()) {
+				body_arguments args{
+					{"token", bearer_token},
+					{"token_type_hint", "access_token"}};
+
+				auto json_res{hit_intro_endpoint(url_encode_body(args))};
+				log::debug("{}: Got the following resp back: [{}]", __func__, json_res);
+
+				// Do fun bearer token validation
+				if (!json_res.at("active").get<bool>()) {
+					log::error("{}: This BOZO thinks they can just waltz on in...", __func__);
+					return {.response = fail(status_type::unauthorized)};
+				}
+
+				// Do mapping of user to irods user
+				const auto& mappings{irods::http::globals::oidc_configuration().at("user_attribute_mapping")};
+
+				// O(n x m)?
```

Comment:
can we create a map and then do a simple lookup?  O(1)?

Reply:
Don't think we can. That would require the JSON in the mapping and the introspection response to exactly match, right?

The introspection response can return some 'time' data, so it won't be consistent.

Reply:
Correct me if I'm wrong, but the reason for the O(n * m) is because the mapping supports identifying a user through a combination of attributes/properties.

If the mapping was strictly something like the following, where `<oidc_atrribute_value>` is the value mapped to `email_address`, then we could get O(1). This is not as flexible as the what @MartinFlores751 has presented.
```js
{
  "<oidc_atrribute_value>": "alice",
  "<oidc_atrribute_value>": "rods"
  // ... more mappings ...
}
```

However, the style of mapping could also be an option. Anything's possible, it's just depends on the pros/cons.

Reply:
Makes sense.   Carry on...